src/JustEat.StatsD/TimerExtensions.cs

```diff
@@ -79,74 +94,88 @@ public static void Time(this IStatsDPublisher publisher, string bucket, Action<I
             }
         }
 
-
         /// <summary>
         /// Starts a new timer for the specified bucket which is started when the specified delegate is invoked
         /// and awaited, and is stopped and published when the asynchronous delegate invocation completes.
         /// </summary>
         /// <param name="publisher">The <see cref="IStatsDPublisher"/> to publish with.</param>
         /// <param name="bucket">The bucket to publish the timer for.</param>
         /// <param name="action">A delegate to a method whose invocation should be timed.</param>
+        /// <param name="tags">The tags to publish with the timer.</param>
         /// <returns>
         /// A <see cref="Task"/> representing the asynchronous operation to time.
         /// </returns>
         /// <exception cref="ArgumentNullException">
         /// <paramref name="publisher"/>, <paramref name="bucket"/> or <paramref name="action"/> is <see langword="null"/>.
         /// </exception>
-        public static async Task Time(this IStatsDPublisher publisher, string bucket, Func<IDisposableTimer, Task> action)
+        public static async Task Time(
+            this IStatsDPublisher publisher,
+            string bucket,
+            Func<IDisposableTimer, Task> action,
+            IDictionary<string, string?>? tags = null)
         {
             if (action == null)
             {
                 throw new ArgumentNullException(nameof(action));
             }
 
-            using (var timer = StartTimer(publisher, bucket))
+            using (var timer = StartTimer(publisher, bucket, tags))
             {
                 await action(timer).ConfigureAwait(false);
             }
         }
 
         /// <summary>
-        /// Starts a new timer for the specified bucket which is started when the specified delegate is invoked
+        /// Starts a new timer for the specified bucket and tags which is started when the specified delegate is invoked
         /// and is stopped and published when the delegate invocation completes.
         /// </summary>
         /// <typeparam name="T">The type of the result of the delegate to invoke.</typeparam>
         /// <param name="publisher">The <see cref="IStatsDPublisher"/> to publish with.</param>
         /// <param name="bucket">The bucket to publish the timer for.</param>
         /// <param name="func">A delegate to a method whose invocation should be timed and result returned.</param>
+        /// <param name="tags">The tags to publish with the timer.</param>
         /// <returns>
         /// The value from invoking <paramref name="func"/>.
         /// </returns>
         /// <exception cref="ArgumentNullException">
-        /// <paramref name="publisher"/>, <paramref name="bucket"/> or <paramref name="func"/> is <see langword="null"/>.
+        /// <paramref name="publisher"/>, <paramref name="bucket"/>, <paramref name="tags"/> or <paramref name="func"/> is <see langword="null"/>.
```

Comment:
Does it throw if `tags` is `null`? The code I've seen so far in the review just seemed to skip tags if the value was null.

Reply:
No, you're right. I forgot to remove it. I will do it.

Reply:
Done!